extensions/warp-ipfs/src/store/community.rs

```diff
@@ -0,0 +1,389 @@
+        Err(Error::Unimplemented)
+    }
+}
+impl CommunityDocument {
+    pub fn new(keypair: &Keypair, name: String) -> Result<Self, Error> {
+        let creator = keypair.to_did()?;
+
+        let mut permissions = CommunityPermissions::new();
+        permissions.insert(CommunityPermission::EditInfo, IndexSet::new());
+        permissions.insert(CommunityPermission::ManageRoles, IndexSet::new());
+        permissions.insert(CommunityPermission::ManagePermissions, IndexSet::new());
+        permissions.insert(CommunityPermission::ManageMembers, IndexSet::new());
+        permissions.insert(CommunityPermission::ManageChannels, IndexSet::new());
+        permissions.insert(CommunityPermission::ManageInvites, IndexSet::new());
+        permissions.insert(CommunityPermission::DeleteMessages, IndexSet::new());
+
+        let mut members = IndexSet::new();
+        members.insert(creator.clone());
+
+        let mut document = Self {
+            id: Uuid::new_v4(),
+            name,
+            description: None,
+            creator,
+            created: Utc::now(),
+            modified: Utc::now(),
+            members,
+            channels: IndexMap::new(),
+            roles: IndexMap::new(),
+            permissions,
+            invites: IndexMap::new(),
+            deleted: false,
+            signature: None,
+        };
+        document.sign(keypair)?;
+        Ok(document)
+    }
+}
+impl From<CommunityDocument> for Community {
+    fn from(value: CommunityDocument) -> Self {
+        let mut community = Community::default();
+        community.set_id(value.id);
+        community.set_name(value.name);
+        community.set_description(value.description);
+        community.set_creator(value.creator);
+        community.set_created(value.created);
+        community.set_modified(value.modified);
+        community.set_members(value.members);
+        community.set_channels(
+            value
+                .channels
+                .iter()
+                .map(|(k, _)| RoleId::parse_str(k).expect("should be valid uuid"))
+                .collect(),
+        );
+        community.set_roles(
+            value
+                .roles
+                .iter()
+                .map(|(k, _)| RoleId::parse_str(k).expect("should be valid uuid"))
+                .collect(),
+        );
+        community.set_permissions(value.permissions);
+        community.set_invites(
+            value
+                .invites
+                .iter()
+                .map(|(k, _)| RoleId::parse_str(k).expect("should be valid uuid"))
+                .collect(),
+        );
+        community
+    }
+}
+impl CommunityDocument {
+    pub fn participants(&self) -> IndexSet<DID> {
+        let mut participants = self.members.clone();
+        participants.insert(self.creator.clone());
+        for (_, invite) in &self.invites {
+            if let Some(target) = &invite.target_user {
+                participants.insert(target.clone());
```

Comment:
It might be worth having a separate topic where updates are sent out instead so subscribers but also been invited would be able to see the updates there. This could probably be done in a separate PR